src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Members.cs

```diff
@@ -410,7 +419,8 @@ public override void VisitMethod(IMethodSymbol symbol)
                         }
                         AddPropertyNameAndParameters(associatedProperty);
                         AddPunctuation(SyntaxKind.DotToken);
-                        AddKeyword(symbol.MethodKind == MethodKind.PropertyGet ? SyntaxKind.GetKeyword : SyntaxKind.SetKeyword);
+                        AddKeyword(symbol.MethodKind == MethodKind.PropertyGet ? SyntaxKind.GetKeyword :
+                            IsInitOnly(associatedProperty) ? SyntaxKind.InitKeyword : SyntaxKind.SetKeyword);
```

Comment:
>IsInitOnly(associatedProperty) [](start = 28, length = 30)

It feels like we should have a helper that takes IMethodSymbol. We have a method at hand, why go back to the property?